plugins/inputs/snmp/snmp.go

```diff
@@ -623,6 +623,11 @@ func (s *Snmp) getConnection(idx int) (snmpConnection, error) {
 	gs := gosnmpWrapper{&gosnmp.GoSNMP{}}
 	s.connectionCache[idx] = gs
 
+	if strings.HasPrefix(agent, "tcp:") {
```

Comment:
Let's use `tcp://` format instead of adopting the net-snmp style.  I think it will provide more consistency with other plugins.